### ISSUE TITLE
Accessibility - Drupal - Focus elements

### ISSUE THREAD

**Issue by adam3smith** [QualitativeDataRepository/dataverse#100]
The visual focus of elements (i.e. how they're highlighted when tabbing through) is very unequal on QDR:
1. Ideal: customized focus on some elements like hamburger menu: 
![image](https://user-images.githubusercontent.com/128986/135639744-ac29b4d5-88b2-473c-aedc-9362188050ae.png)

2. Not ideal: standard, thin dotted line focus, e.g., for Qualitative Data Repository
![image](https://user-images.githubusercontent.com/128986/135639859-652ff1ce-f824-481b-bab3-93ffd15ed250.png)

3. Not acceptable at all: No visible focus, e.g. on blue "Discover Data" and "Deposit Data" buttons

Ideally we'd see some version of 1. throughout. Some useful resources:
https://www.deque.com/blog/accessible-focus-indicators/
https://usability.yale.edu/web-accessibility/articles/focus-keyboard-operability

**Comment by qqmyers** [comment]
This appears to be ~browser-specific - with Firefox using blue and dotted lines (but not for buttons), while Chrome and Edge use a solid black box. (Not sure what Safari does). I can do a relatively simple override of everything, e.g. adding
```
:focus-visible {
  outline: 1px solid black;
}
```
in the QDR theme's styles.css file (which is loaded for the header and so gets Drupal and Dataverse pages) - a change that's currently on dev. I'm not sure how yet, but I think there are ways to be more browser-selective, and these changes can certainly be made element specific if that's useful (e.g. so buttons have a different highlight than other elements as Firefox's default), we can adjust the margin, etc.

**Comment by adam3smith** [comment]
Interesting -- yeah, we'd want
- [ ] the Firefox behavior to mimic Edge/Chrome's
- [ ] the buttons (like Discover and Deposit) to have some space around the focus box so it's visible
- [ ] similar behavior on DV, where focus is really hard to see

**Comment by qqmyers** [comment]
I've made some css changes in Drupal that force a thin blue box for the focus, adding some padding that makes it work for links and makes it visible on blue buttons. Let me know what you think. 

One thing I haven't found how to fix yet is to change the no-op tab stop on checkboxes in the Dataverse facets. The initial tabstop, which allows you to select/unselect that checkbox has a blue focus box, the next tabstop that does nothing if you hit enter, has a black outline with no margin, and then the third stop on the name, which selects that item exclusively, have the blue box. The fact that there's a tabstop in the middle at all is a PrimeFaces quirk (in some other issue), but I'm not sure what css would change the focus box (or if that's useful - since there's no action for it).

Beyond that, if you see any elements that don't have the blue focus box or a reasonable custom focus already, let me know and I can look in the css to try fixing it.

Lastly, right now I've installed two separate Javascripts that attempt to highlight where the focus is, either animating it moving between elements (Dataverse side) or starting the focus box slightly larger and shrinking to the final size (Drupal side). These come from https://github.com/NV/flying-focus . Out of the box, they don't appear to work on Firefox, but do on Chrome and Edge on a PC. If either form is worth keeping, let me know and I can look into the firefox issue - my guess is that is is related to the script running before all the DOM elements are in place. If not, I can remove and deploy things to stage. (Without the scripts, the focus just appears immediately in final form at the new element.)

**Comment by qqmyers** [comment]
FYI - the update I just pushed probably removed the javascripts for focus animation - let me know if you want them restored.

**Comment by adam3smith** [comment]
OK, the checkbox behavior isn't ideal, but we'll label this upstream, I guess? Looking great otherwise
